test/async-hooks/test-callback-error.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const spawnSync = require('child_process').spawnSync;
+const async_hooks = require('async_hooks');
+const initHooks = require('./init-hooks');
+
+switch (process.argv[2]) {
+  case 'test_init_callback':
+    initHooks({
+      oninit: common.mustCall(() => { throw new Error('test_init_callback'); })
```

Comment:
There's the problem where if `test_init_callback` is never hit then the `common.mustCall()` will never run. Instead you'll need to do it like this:
```js
const oninitMustCall = common.mustCall(() => { throw new Error('test_init_callback'); });
switch (process.argv[2]) {
  case 'test_init_callback':
    initHooks({ oninit: oninitMustCall }).enable();
```